raiden/tests/utils/transport.py

```diff
@@ -1,4 +1,225 @@
+
+        user_addr_hex = user_match.group(1)
+        user_addr = unhexlify(user_addr_hex[2:])
+
+        rec_addr = eth_recover(data=self.hs_hostname.encode(), signature=signature)
+        if not rec_addr or rec_addr != user_addr:
+            self.log.error(
+                'invalid account password/signature. user=%r, signer=%r',
+                user_id,
+                rec_addr,
+            )
+            defer.returnValue(False)
+
+        localpart = user_id.split(":", 1)[0][1:]
+        self.log.info('eth login! valid signature. user=%r', user_id)
+
+        if not (yield self.account_handler.check_user_exists(user_id)):
+            self.log.info('first user login, registering: user=%r', user_id)
+            yield self.account_handler.register(localpart=localpart)
+
+        defer.returnValue(True)
+
+    @staticmethod
+    def parse_config(config):
+        return config
+
+
+def make_requests_insecure():
+    """
+    Prevent `requests` from performing TLS verification.
+
+    **THIS MUST ONLY BE USED FOR TESTING PURPOSES!**
+    """
+    # Disable verification in requests by replacing the 'verify'
+    # attribute with non-writable property that always returns `False`
+    requests.Session.verify = property(lambda self: False, lambda self, val: None)
+
+
+@contextmanager
+def generate_synapse_config() -> ContextManager:
+    # Allows caching of self signed synapse certificates on CI systems
+    if _SYNAPSE_BASE_DIR_VAR_NAME in os.environ:
+        synapse_base_dir = Path(os.environ[_SYNAPSE_BASE_DIR_VAR_NAME])
+        synapse_base_dir.mkdir(parents=True, exist_ok=True)
+        delete_base_dir = False
+    else:
+        synapse_base_dir = Path(mkdtemp(prefix='pytest-synapse-'))
+        delete_base_dir = True
+
+    def generate_config(port: int):
+        server_dir = synapse_base_dir.joinpath(f'localhost-{port}')
+        server_dir.mkdir(parents=True, exist_ok=True)
+
+        server_name = f'localhost:{port}'
+
+        # Always overwrite config file to ensure we're not using a stale version
+        config_file = server_dir.joinpath('synapse_config.yaml').resolve()
+        config_template = _SYNAPSE_CONFIG_TEMPLATE.read_text()
+        config_file.write_text(config_template.format(server_dir=server_dir, port=port))
+
+        tls_key_file = server_dir.joinpath(f'{server_name}.tls.crt')
+
+        if not tls_key_file.exists():
+            subprocess.run(
+                [
+                    sys.executable,
+                    '-m',
+                    'synapse.app.homeserver',
+                    f'--server-name={server_name}',
+                    f'--config-path={config_file!s}',
+                    '--generate-keys',
+                ],
+                cwd=server_dir,
+                timeout=30,
+                check=True,
+                stderr=subprocess.DEVNULL,
+                stdout=subprocess.DEVNULL,
+            )
+        return server_name, config_file
+
+    yield generate_config
```

Comment:
I'd recommend putting `if` below under a `tr/finally` block wrapping `yield`, or else raised exceptions inside context will make `shutil.rmtree` not be reached, which may aggregate on quite big directories sometimes

Reply:
Or is it intended?

Reply:
Good catch, thanks.
It wasn't intended. Wouldn't have been a problem on CI since there the config dir is kept via cache anyway to avoid re-generating the certificates every time but still it's cleaner.